src/frontend/src/optimizer/plan_node/logical_iceberg_scan.rs

```diff
@@ -102,19 +132,277 @@ impl ExprRewritable for LogicalIcebergScan {}
 impl ExprVisitable for LogicalIcebergScan {}
 
 impl PredicatePushdown for LogicalIcebergScan {
+    /// NOTE(kwannoel):
+    /// 1. We expect it to be constant folded
+    /// 2. We don't convert `inputRefs` of type boolean directly to `IcebergPredicates`.
+    /// 3. The leaf nodes are always logical comparison operators:
+    ///    `Equal`, `NotEqual`, `GreaterThan`,
+    ///    `GreaterThanOrEqual`, `LessThan`, `LessThanOrEqual`.
+    /// 4. For leaf nodes, their LHS is always an `inputRef`
+    ///    and their RHS is always a `Literal` to be compatible with Iceberg.
     fn predicate_pushdown(
         &self,
         predicate: Condition,
         _ctx: &mut PredicatePushdownContext,
     ) -> PlanRef {
+        fn rw_literal_to_iceberg_datum(literal: &Literal) -> Option<IcebergDatum> {
```

Comment:
Thinking if it's better to add a separate optimization pass instead, at the end of batch, to pattern match:
```
BatchFilter -> BatchIcebergScan
```

And rewrite both of them, to push down the predicate inside `BatchFilter` to `BatchIcebergScan`.

wdyt @chenzl25

Reply:
It's mainly for future-proofing, in case we support batch share plan. Then the predicate inside `BatchIcebergScan` will always be empty during the share plan optimization pass, so we won't need to compare against it.

And the predicate will be held all the way by `BatchFilter`, as a rw predicate. That way only after any share plan optimization pass in batch, then we push down the rw predicate to the iceberg one, in a single step.

Reply:
Changed it: https://github.com/risingwavelabs/risingwave/pull/19228/commits/115e2ef8dcbfab47653ae5103d17dff74b906942. I think it makes more sense.

Besides preserving the semantics of `Eq` and `Hash` for `Batch Share Plan` (if any in the future), we also avoid multiple pushdown passes just for iceberg. We can just do it in one shot at the end, and benefit from other optimization pass like constant folding of RW.

Reply:
It looks good to me to put predicate pushdown at the end of the batch optimization at the current stage. However, in the future, I think unifying all predicate pushdown in a logic operator will make more sense because we have a predicate pushdown trait in a logic operator.

Reply:
> It's mainly for future-proofing, in case we support batch share plan

This is the main reason. Otherwise we have to support a full-IR for iceberg predicates, so we can properly support `Eq` and `Hash`.